src/realm/util/file.cpp

```diff
@@ -1610,6 +1610,30 @@ void File::MapBase::sync()
     File::sync_map(m_fd, m_addr, m_size);
 }
 
+std::time_t File::last_write_time(const std::string& path)
+{
+#ifdef _WIN32
+    int fd, result;
+    struct _stat buf;
+    fd = _open(path.c_str(), _O_RDONLY);
+    if (fd == -1) {
+        throw std::system_error(errno, std::system_category(), "_sopen_s() failed");
+    }
+    result = _fstat(fd, &buf);
+    if (result != 0) {
+        _close(fd);
+        throw std::system_error(errno, std::system_category(), "_fstat() failed");
+    }
+    return buf.st_mtime;
+#else // POSIX version
+
+    struct stat statbuf;
+    if (::stat(path.c_str(), &statbuf) == 0) {
```

Comment:
Why is the success/failure branching reversed vs above? I think `if (failure) throw_or_return` is more idiomatic.